zenoh-kotlin/src/commonTest/kotlin/io/zenoh/KeyExprTest.kt

```diff
@@ -52,10 +52,10 @@ class KeyExprTest {
         assertNotEquals(keyExpr1, keyExpr3)
 
         keyExpr2.close()
-        assertNotEquals(keyExpr1, keyExpr2)
+        assertEquals(keyExpr1, keyExpr2)
 
         keyExpr1.close()
-        assertNotEquals(keyExpr1, keyExpr2)
+        assertEquals(keyExpr1, keyExpr2)
```

Comment:
is this logic flip intentional ?

Reply:
Yes, previously when the key expression was closed (meaning it wasn't bound anymore to a native keyexpr) then the equals operation returned false, since it was done at native level. Now, no matter if the key expression was closed, the equals is done by comparing the string representation of the key expression, that is a constant value associated to the keyexpr object.